### PR TITLE
[4369] add query limit to query_processor (4-2-stable)

### PR DESCRIPTION
**Pull request by jasoncoposky**

### CODE REVIEW

**Comment by trel** [comment]
noob question... is this all it needs?   where is it handled?

**Comment by jasoncoposky** [comment]
yes, it is passed through to the query iterator which [enforces the limit](https://github.com/irods/irods/blob/master/lib/core/include/irods_query.hpp#L84).

**Comment by trel** [comment]
easy enough

**Comment by trel** [comment]
i note that limit is `uint32_t`, and this one is `uintmax_t`

**Comment by jasoncoposky** [comment]
fair.

File: lib/core/include/query_processor.hpp [hunk 1]
```diff
@@ -69,6 +70,7 @@ namespace irods
     private:
         std::string query_;
         job job_;
+        uintmax_t limit_;
```
Comment (trel) [inline lib/core/include/query_processor.hpp]: still uintmax_t

**Pull request merged**